src/extensions.jl

```diff
@@ -86,7 +86,7 @@ tags[:array] = d ->
 isprimitive(T) = fieldcount(T) == 0 && T.size > 0
 
 structdata(x) = isprimitive(typeof(x)) ? reinterpret_(UInt8, [x]) :
-    Any[getfield(x, f) for f in fieldnames(typeof(x))]
+    Any[getfield(x, f) for f in fieldnames(typeof(x)) if isdefined(x, f)]
```

Comment:
we can potentially run into issues here when trying to construct back the types, right?

Reply:
If the field is uninitialized at this point, then there must be a constructor that excludes that field from its args right?

Reply:
I think this should be fine because that argument would just not be there on re-construction. Which is fine cause how would it be uninitialized otherwise?

Reply:
Constructors don't in general map to the fields. The set of isdefined fields is frequently also not contiguous.